### PR TITLE
update upgrade on inv move

### PR DESCRIPTION
**Pull request by SwissalpS**
As suggested by Huhhila.
Have not tested, just pasted in as given and formated

### CODE REVIEW

**Review by BuckarooBanzay — APPROVED** [review]

**Comment by SwissalpS** [comment]
It's only sometimes working. Moving an upgrade from 'books inventory' to upgrades and vice versa doesn't always update the tool-tip of the engine.

**Comment by SwissalpS** [comment]
should probably also add permission check for inventory movements

**Comment by SwissalpS** [comment]
I just realized the infotext is updated on technic run procedure, so there is a delay.

**Comment by SwissalpS** [comment]
worked for me. Also patches the loophole where players were theoretically able to extract/put items of a protected jd

File: engines/default.lua [hunk 1]
```diff
@@ -204,20 +204,29 @@ minetest.register_node("jumpdrive:engine", {
 	end,
 
 	-- inventory protection
+	allow_metadata_inventory_move = function(pos, _, _, _, _, count, player)
+		if (not player) -- no player given
+			or (not player:is_player()) -- not a player
+			or minetest.is_protected(pos, player:get_player_name()) -- protected
+		then return 0 end
```
Comment (OgelGames) [inline engines/default.lua:211]: Parentheses and comments are not necessary.
 ```suggestion
		if not player or not player:is_player() or minetest.is_protected(pos, player:get_player_name()) then
			return 0
		end
```
Not sure if `player:is_player()` check is really needed either 🤔

File: engines/default.lua [hunk 2]
```diff
@@ -204,20 +204,29 @@ minetest.register_node("jumpdrive:engine", {
...
+
 	allow_metadata_inventory_take = function(pos, listname, index, stack, player)
-		if player and player:is_player() and minetest.is_protected(pos, player:get_player_name()) then
-			-- protected
-			return 0
-		end
+		if (not player) -- no player given
+			or (not player:is_player()) -- not a player
+			or minetest.is_protected(pos, player:get_player_name()) -- protected
+		then return 0 end
```
Comment (OgelGames) [inline engines/default.lua:220]:
```suggestion
			if not player or not player:is_player() or minetest.is_protected(pos, player:get_player_name()) then
				return 0
			end
```

File: engines/default.lua [hunk 3]
```diff
@@ -204,20 +204,29 @@ minetest.register_node("jumpdrive:engine", {
...
 
 	allow_metadata_inventory_put = function(pos, listname, index, stack, player)
-		if player and player:is_player() and minetest.is_protected(pos, player:get_player_name()) then
-			-- protected
-			return 0
-		end
+		if (not player) -- no player given
+			or (not player:is_player()) -- not a player
+			or minetest.is_protected(pos, player:get_player_name()) -- protected
+		then return 0 end
```
Comment (OgelGames) [inline engines/default.lua:229]:
```suggestion
			if not player or not player:is_player() or minetest.is_protected(pos, player:get_player_name()) then
				return 0
			end
```

File: engines/default.lua (hunk 1 above)
Comment (SwissalpS) [reply engines/default.lua:211]: yes, I'll remove the comments. The parenthis are not needed but they help ensure that the reader understands what is going on in chained and/or checks. I consider it a maintenance speed up.

I think the is_player check is ok. Some server may have a 'thief' mob installed that tries to take stuff from inventories.
Or similar.

**Review by BuckarooBanzay — APPROVED** [review]

**Comment by BuckarooBanzay** [comment]
LGTM :+1:

**Comment by BuckarooBanzay** [comment]
ugh, didn't squash the commits :shrug:

**Comment by OgelGames** [comment]
Fixed it: https://github.com/mt-mods/jumpdrive/commit/631879bcbecb489b8e92fcd2e6f18ccf857bc6b5 :)

**Comment by BuckarooBanzay** [comment]
thanks :)

**Pull request merged**